src/stactools/naip/utils.py

```diff
@@ -5,7 +5,9 @@
 
 import dateutil.parser
 
-NAIP_FILENAME_REGEX = re.compile(r"(m)_(\d{7})_(\w{2})_(\d{2})_(\d{1,3})_(\d{8})")
+NAIP_FILENAME_REGEX = re.compile(
+    r"m_(?:\d{7})_(?:\w{2})_(?:\d{2})_(?:\w{1,3})_(?P<dt>\d{8})(?:_\d{8})?"
```

Comment:
I think the only one of the non-capturing groups that is needed is `(?:_\d{8})?` because it's optionally capturing the underscore and the digits -- all of the other non-capturing groups are single units and don't need to be grouped at all.

Reply:
updated